### PR TITLE
chore: fine-tune error and help messages of CLI commands for slot caps

### PR DESCRIPTION
**Pull request by amandavialva01**
## Ticket
DET-10438

## Description
Fine-tune the help and error messages returned by some of the CLI commands so that the user has a clearer idea of mututally exclusive commands as well as a clearer direction of how to modify an incorrectly formatted command.

## Test Plan
The following commands were modified, so run these on both single and MultiRM clusters and verify a clear and indicative response:
- Help Message
  - The `--cluster-name` flag for `det w create --help` and `det w bindings set --help`
- Error Message
  - `det w bindings set <workspace_name>` with no specified namespace or auto-created namespace
    - Added e2e test for both single and multirm
  - `det w create <workspace_name> --auto-create-namespace-all-clusters --resource-quota <resource_quota>`
    - Added e2e test for both single and multirm (should fail for multiRM and pass for singleRM)
  - `det w resource-quota set <workspace_name> <quota>` when the requested quota is higher than an already existing quota placed on the namespace
    - Slight improvements here, mostly just added the name of the lower existing quota into the stdout and tweaked one of the integration test results that verifies the request fails
  - `det w create <workspace_name> --auto-create-namespace-all-clusters --cluster-name` and `det w bindings set <workspace_name> --auto-create-namespace-all-clusters --cluster-name`
    - Added e2e tests for both single and multirm (should fail for both)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: harness/determined/cli/workspace.py [hunk 1]
```diff
@@ -210,18 +210,24 @@ def list_workspace_namespace_bindings(args: argparse.Namespace) -> None:
...
+        remove_cluster_name = ""
+        if args.cluster_name:
+            remove_cluster_name = "remove --cluster-name CLUSTER_NAME and "
+
         raise api.errors.BadRequestException(
-            "must provide --namespace NAMESPACE or --auto-create-namespace, or remove "
-            + "--cluster-name CLUSTER_NAME and specify --auto-create-namespace-all-clusters"
+            "must provide --namespace NAMESPACE or --auto-create-namespace, or "
+            + remove_cluster_name
+            + "specify --auto-create-namespace-all-clusters"
```
Comment (jgongd) [inline harness/determined/cli/workspace.py:223]: I like this updated message!

File: master/internal/api_workspace.go [hunk 2]
```diff
@@ -830,11 +845,10 @@ func (a *apiServer) setWorkspaceNamespaceBindings(ctx context.Context,
 ) (resp *apiv1.SetWorkspaceNamespaceBindingsResponse, err error) {
 	defer func() {
 		if v := recover(); v != nil {
-			err = status.Error(codes.InvalidArgument, "Auto-creating namespaces is "+
+			err = status.Errorf(codes.InvalidArgument, "Auto-creating namespaces is "+
```
Comment (NicholasBlaskey) [inline master/internal/api_workspace.go:848]: does ```Error``` vs ```Errorf``` here matter since we aren't specifying formatting args?

**Review by NicholasBlaskey — APPROVED** [review]
lgtm

**Review by jgongd — APPROVED** [review]

File: master/internal/api_workspace.go (hunk 2 above)
Comment (amandavialva01) [reply master/internal/api_workspace.go:848]: Good catch, changed!

File: e2e_tests/tests/cluster/test_wksp_nmsp_concurrency.py [hunk 3]
```diff
@@ -14,8 +14,9 @@
 # a thread that locks the jobsService tries to reacquire the lock during the execution of a callback
 # function that only gets called when a job is running in the namespace that we want to bind to the
 # workspace. Verify that we don't run into this deadlock when triggering multiple calls to the
-# API handler that binds a workspace to an auto-created namespace, as this request can trigger the
-# deadlock if the namespace (or verify the existence of for the first time) is running a job.
+# API handler that binds a workspace to an auto-created namespace (or verifies the existence of a
+# namespace for the first time), as this request can trigger the deadlock if the namespace is
+# running a job.
```
Comment (tara-det-ai) [inline e2e_tests/tests/cluster/test_wksp_nmsp_concurrency.py:19]: is there a better way to say "as this request can" - this phrase can be confusing for some users

Comment (tara-det-ai) [reply e2e_tests/tests/cluster/test_wksp_nmsp_concurrency.py:19]: maybe

# API handler that binds a workspace to an auto-created namespace or verifies the existence of a namespace.
# This request can trigger a deadlock if the namespace is running a job.

**Review by corban-beaird — APPROVED** [review]
Looks solid!

File: e2e_tests/tests/cluster/test_wksp_nmsp_concurrency.py (hunk 3 above)
Comment (amandavialva01) [reply e2e_tests/tests/cluster/test_wksp_nmsp_concurrency.py:19]: @tara-det-ai , this comment is mostly so devs looking at this end-to-end test can understand its purpose (so shouldn't have any front-facing impact).
Nevertheless, I'll remove the "as this request can" phrase!

**Pull request merged**